pallets/pallet-deposit-storage/src/fungible/try_state.rs

```diff
@@ -0,0 +1,66 @@
+// KILT Blockchain – https://botlabs.org
+// Copyright (C) 2019-2024 BOTLabs GmbH
+
+// The KILT Blockchain is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// The KILT Blockchain is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+// If you feel like getting in touch with us, you can do so at info@botlabs.org
+
+use frame_support::{ensure, traits::fungible::InspectHold};
+use frame_system::pallet_prelude::BlockNumberFor;
+use kilt_support::Deposit;
+use sp_runtime::{traits::CheckedAdd, TryRuntimeError};
+use sp_std::collections::btree_map::BTreeMap;
+
+use crate::{deposit::DepositEntry, AccountIdOf, BalanceOf, Config, HoldReason, SystemDeposits};
+
+// Verify the state kept as part of the `MutateHold` implementation is
+// consistent with the state of the `Currency` type.
+pub(crate) fn check_fungible_consistency<T>(_n: BlockNumberFor<T>) -> Result<(), TryRuntimeError>
+where
+	T: Config,
+{
+	// Sum together all the deposits stored as part of the `MutateHold`
+	// implementation, and fail if any of them does not have the expected
+	// `crate::HoldReason::FungibleImpl` reason.
+	let sum_of_deposits = SystemDeposits::<T>::iter_values().try_fold(
+		BTreeMap::<AccountIdOf<T>, BalanceOf<T>>::new(),
+		|mut sum,
+		 DepositEntry {
+		     reason,
+		     deposit: Deposit { amount, owner },
+		 }| {
+			ensure!(
+				reason == HoldReason::FungibleImpl.into(),
+				TryRuntimeError::Other("Found a deposit reason different than the expected `HoldReason::FungibleImpl`")
+			);
+
+			// Fold the deposit amount for the current user.
+			let entry = sum.entry(owner).or_default();
+			*entry = entry.checked_add(&amount).expect("Failed to fold deposits for user.");
+
+			Ok::<_, TryRuntimeError>(sum)
```

Comment:
I kinda prefer a function definition then the type injection here | sum, DepositEntry {. . }| -> Result<BtreeMap<..>, TryRuntimeError>. But it is a minor thing.

Reply:
https://github.com/KILTprotocol/kilt-node/pull/830/commits/f9d2aa7af4c8dda7c250f5a0442bfdccf1b7b2cd

Reply:
Ops, I forgot it in the other function: https://github.com/KILTprotocol/kilt-node/pull/830/commits/cee766cd9e20f814cc2b6535fa1cb789719839f0